### ISSUE TITLE
Challenging Example

### ISSUE THREAD

**Issue by ffrohn** [draperlaboratory/cage-koat#2]
The following example is an abstraction of an array algorithm. The second argument `i` of `f` corresponds to the integer which is used as index. The first argument of `f` is the size of the array where all elements at positions `j` with `j < i` are disregarded (i.e., only the 'subarray' starting at position `i` is considered). The third argument `n` is the length of the array.

`(GOAL COMPLEXITY)`
`(STARTTERM (FUNCTIONSYMBOLS start))`
`(VAR x)`
`(RULES`
`start(x) -> f(x, 0, n) :|: x > 0 && 0 <= n && n < x`
`f(x, i, n) -> g(z, y, i+1, n) :|: y >= 0 && z >= 0 && y <= x && z <= x && x = z + y && i < n`
`g(x, y, i, n) -{y,y}> f(x, i, n)`
`)`

The second rule reads the `i`-th element `y` from the array, increments `i` (s.t. the new size of the considered subarray is `x - y`) and calls `g`. The function `g` is linear in `y` and calls `f`.

In short: `f` invokes a function with linear complexity for each element of the array. Hence, the runtime of the ITS is linear. Unfortunately, KoAT just proves a quadratic bound. As this is a pretty common pattern, we should probably have a look at it in more detail at some point.

PS.: Are the Issues from the old repository still available somewhere?
